jmix-ui/ui/src/main/java/io/jmix/ui/component/filter/FilterMetadataTools.java

```diff
@@ -98,6 +99,141 @@ protected List<MetaPropertyPath> getPropertyPaths(MetaClass filterMetaClass,
         return paths;
     }
 
+    public List<MetaPropertyPath> getPropertyPathsFromIncludedProperties(MetaClass filterMetaClass,
+                                                                         String query,
+                                                                         @Nullable Predicate<MetaPropertyPath> propertiesFilterPredicate,
+                                                                         List<String> includedProperties) {
+        Node<String> includedPropertiesTree = new Node<>("");
+
+        List<MetaPropertyPath> paths = new ArrayList<>();
+
+        for (String property : includedProperties) {
+            List<String> pathStrings = Arrays.asList(property.split("\\."));
+            recursivelyAddPropertyToIncludedPropertiesTree(filterMetaClass, includedPropertiesTree, pathStrings, filterMetaClass, query);
+        }
+
+        List<String> propertyPaths = getAllPathsFromTree(includedPropertiesTree);
+        for (String propertyPath : new ArrayList<>(new HashSet<>(propertyPaths))) {
```

Comment:
Duplicate paths may be present. The List -> Set -> List conversion is to remove these duplicates.